### PR TITLE
refactor(material/datepicker): Remove use of zone onStable for focus and dropdown positioning

### PR DESCRIPTION
**Pull request by mmalerba**
passing TGP

### CODE REVIEW

File: src/material/datepicker/calendar-body.ts [hunk 1]
```diff
@@ -310,21 +313,24 @@ export class MatCalendarBody<D = any> implements OnChanges, OnDestroy, AfterView
    */
   _focusActiveCell(movePreview = true) {
     this._ngZone.runOutsideAngular(() => {
```
Comment (atscott) [inline src/material/datepicker/calendar-body.ts:315]: I don't think you need `runOutsideAngular` anymore. `afterRender*` callbacks run outside the zone already.

File: src/material/datepicker/calendar-body.ts [hunk 2]
```diff
@@ -309,8 +312,8 @@ export class MatCalendarBody<D = any> implements OnChanges, OnDestroy, AfterView
    * Adding delay also complicates writing tests.
    */
   _focusActiveCell(movePreview = true) {
-    this._ngZone.runOutsideAngular(() => {
-      this._ngZone.onStable.pipe(take(1)).subscribe(() => {
+    afterNextRender(
+      () => {
         setTimeout(() => {
```
Comment (crisbeto) [inline src/material/datepicker/calendar-body.ts:317]: Could we get rid of the `setTimeout` here?

Comment (mmalerba) [reply src/material/datepicker/calendar-body.ts:317]: I originally tried it without the setTimeout and the TGP had failures, so I put it back

**Review by wagnermaciel — APPROVED** [review]

**Review by zarend — COMMENTED** [review]
👍

**Pull request merged**